bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/BusinessPartnerChangesDetector.kt

```diff
@@ -0,0 +1,259 @@
+            return true
+        }
+
+        return originalSet.any { originalAddress ->
+            val updatedAddress = updatedSet.find { it == originalAddress }
+            updatedAddress == null || haveLogisticAddressChanges(originalAddress, updatedAddress)
+        }
+    }
+
+
+    fun haveLogisticAddressChanges(original: LogisticAddress, updated: LogisticAddress): Boolean {
+        return (
+                original.bpn != updated.bpn ||
+                        original.name != updated.name ||
+                        havePhysicalPostalAddressChanges(original.physicalPostalAddress, updated.physicalPostalAddress) ||
+                        haveAlternativePostalAddressChanges(original.alternativePostalAddress, updated.alternativePostalAddress)
+                )
+    }
+
+
+    fun haveLogisticAddressSiteChanges(original: LogisticAddress, updated: LogisticAddress): Boolean {
+        return (
+                original.name != updated.name ||
+                        havePhysicalPostalAddressChanges(original.physicalPostalAddress, updated.physicalPostalAddress) ||
+                        haveAlternativePostalAddressChanges(original.alternativePostalAddress, updated.alternativePostalAddress)
+                )
+    }
+
+
+    fun haveSetSiteChanges(originalSet: Set<Site>, updatedSet: Set<Site>): Boolean {
+        if (originalSet.size != updatedSet.size) {
+            return true
+        }
+
+        return originalSet.any { originalSite ->
+            val updatedSite = updatedSet.find { it == originalSite }
+            updatedSite == null || haveSiteChanges(originalSite, updatedSite)
+        }
+    }
+
+    fun haveSiteChanges(original: Site, updated: Site): Boolean {
+        return (
+                original.bpn != updated.bpn ||
+                        original.name != updated.name ||
+                        haveLogisticAddressSiteChanges(original.mainAddress, updated.mainAddress)
+                )
+    }
+
+    fun haveAlternativePostalAddressChanges(
+        original: AlternativePostalAddress?,
+        updated: AlternativePostalAddress?
+    ): Boolean {
+        return (
+                haveGeographicCoordinatesChanges(original?.geographicCoordinates, updated?.geographicCoordinates) ||
+                        original?.country != updated?.country ||
+                        haveRegionChanges(original?.administrativeAreaLevel1, updated?.administrativeAreaLevel1) ||
+                        original?.postCode != updated?.postCode ||
+                        original?.city != updated?.city ||
+                        original?.deliveryServiceType != updated?.deliveryServiceType ||
+                        original?.deliveryServiceNumber != updated?.deliveryServiceNumber ||
+                        original?.deliveryServiceQualifier != updated?.deliveryServiceQualifier
+                )
+    }
+
+
+    fun haveGeographicCoordinatesChanges(original: GeographicCoordinate?, updated: GeographicCoordinate?): Boolean {
```

Comment:
Instead of naming everying as having changes I would coin this more like "isEquivalent" as that is what you want to check for.

In any case if you want to stick to that name please use singular 'has...Changes'